src/languages/ruby.js

```diff
@@ -80,6 +80,25 @@ export default function(hljs) {
       }
     ]
   };
+
+  // Ruby syntax is underdocumented, but this grammar seems to be accurate
+  // as of version 2.7.2 (confirmed with (irb and `Ripper.sexp(...)`)
+  // https://docs.ruby-lang.org/en/2.7.0/doc/syntax/literals_rdoc.html#label-Numbers
+  var decimal = '[1-9](_?[0-9])*|0';
+  var digits = '[0-9](_?[0-9])*';
+  var NUMBER = {
+    className: 'number', relevance: 0,
+    variants: [
+      // decimal integer/float, optionally exponential or rational, optionally imaginary
+      { begin: `\\b(${decimal})(\\.(${digits}))?([eE][+-]?(${digits})|r)?i?\\b` },
+
+      // 0-prefixed octal integer or explicit decimal/binary/octal/hexadecimal integer,
+      // optionally rational and/or imaginary
+      { begin: "\\b(0[dD][0-9](_?[0-9])*|0[bB][01](_?[01])*|0([oO][0-7])?(_?[0-7])*" +
```

Comment:
So we split this into individual variants for readability?  IE, one for decimal, one for binary, one for octal, one for hex, etc...

Reply:
There's a performance penalty for each additional variant, and Ruby numeric literals are conveniently expressible with just one for float and one for integer. But I could split the sub-patterns into distinct lines joined by string concatenation if you prefer?

Reply:
>There's a performance penalty for each additional variant  

Are you sure? :) I think not - or it's very, very slight.  Internally (inside the parse engine) it all ends up in a `|` divided regex anyways.. but in the grammar individual variants are MUCH simpler to read and understand.

Reply:
OK, fair enough. Separated.